x-pack/osquerybeat/internal/pkgutil/expand.go

```diff
@@ -0,0 +1,118 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package pkgutil
+
+import (
+	"compress/gzip"
+	"errors"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/elastic/beats/v7/x-pack/osquerybeat/internal/cpio"
+	"github.com/elastic/beats/v7/x-pack/osquerybeat/internal/xar"
+)
+
+var ErrPayloadNotFound = errors.New("pkg Payload file not found")
+
+func Expand(pkgFile, dstDir string) error {
```

Comment:
It feels like this and possibly the underlying cpio and xar packages could have tests. 

Adding a simple reference archive to the repository and having a unit test to ensure you can expand it seems like it would cover most of the new functionality.

Reply:
This code is running as a part of the osquerybeat packaging, the successful build itself is a test against the actual osquery.pkg distro.